nifi-api/src/main/java/org/apache/nifi/components/ValidationContext.java

```diff
@@ -99,4 +100,9 @@
      * @return the identifier of the ProcessGroup that the component being validated lives in
      */
     String getProcessGroupIdentifier();
+
+    /**
+     * @return the {@link NodeTypeProvider} which can be used to detect the node type of this NiFi instance.
+     */
+    NodeTypeProvider getNodeTypeProvider();
```

Comment:
I don't think this should be exposed here. The primary node check is inherently dangerous for a validation check and something we shouldn't encourage. Instead let the user get the state in the init and use the clustered value in the validation check.